hive-metastore/src/main/java/org/apache/iceberg/hive/HiveMetadataPreservingTableOperations.java

```diff
@@ -131,6 +136,15 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       boolean tableExists = metaClients.run(client -> client.tableExists(database, tableName));
       if (tableExists) {
         tbl = metaClients.run(client -> client.getTable(database, tableName));
+        String columns = "";
+        try {
+          columns = tbl.getSd().getCols().stream().map(column -> column.getName() + " " + column.getType())
+              .collect(Collectors.joining("\n"));
+        } catch (Throwable throwable) {
+          LOG.debug("Encountered {} while fetching columns for {}.{}", throwable.getMessage(),
+              tbl.getDbName(), tbl.getTableName(), throwable);
+        }
```

Comment:
pull this into a private method for reuse?

Reply:
Exactly what @autumnust said in https://github.com/linkedin/iceberg/pull/62#discussion_r604455966
Resolving this.